src/Microsoft.AspNetCore.Razor/Compilation/TagHelpers/TagHelperDescriptor.cs

```diff
@@ -200,5 +202,23 @@ public IEnumerable<TagHelperRequiredAttributeDescriptor> RequiredAttributes
         /// tag helper.
         /// </summary>
         public TagHelperDesignTimeDescriptor DesignTimeDescriptor { get; set; }
+
+        public Dictionary<string, string> PropertyBag
```

Comment:
Does this need to be settable? In general collection properties (lists, dictionaries etc) aren't settable. It means you don't have to deal with the null checks etc here.